diff -rupN --no-dereference gdal-3.10.1-fedora/frmts/pdf/pdfdataset.cpp gdal-3.10.1-fedora-new/frmts/pdf/pdfdataset.cpp
--- gdal-3.10.1-fedora/frmts/pdf/pdfdataset.cpp	2025-01-08 15:54:06.000000000 +0100
+++ gdal-3.10.1-fedora-new/frmts/pdf/pdfdataset.cpp	2025-02-11 12:05:42.107459437 +0100
@@ -1964,6 +1964,14 @@ CPLErr PDFDataset::ReadPixels(int nReqXO
         PDFDoc *poDoc = m_poDocPoppler;
         poSplashOut->startDoc(poDoc);
 
+        // Note: Poppler 25.2 is certainly not the lowest version where we can
+        // avoid the hack.
+#if !(POPPLER_MAJOR_VERSION > 25 ||                                            \
+      (POPPLER_MAJOR_VERSION == 25 && POPPLER_MINOR_VERSION >= 2))
+#define USE_OPTCONTENT_HACK
+#endif
+
+#ifdef USE_OPTCONTENT_HACK
         /* EVIL: we modify a private member... */
         /* poppler (at least 0.12 and 0.14 versions) don't render correctly */
         /* some PDFs and display an error message 'Could not find a OCG with
@@ -1978,6 +1986,7 @@ CPLErr PDFDataset::ReadPixels(int nReqXO
         OCGs *poOldOCGs = poCatalog->optContent;
         if (!m_bUseOCG)
             poCatalog->optContent = nullptr;
+#endif
         try
         {
             poDoc->displayPageSlice(poSplashOut, m_iPage, m_dfDPI, m_dfDPI, 0,
@@ -1988,14 +1997,19 @@ CPLErr PDFDataset::ReadPixels(int nReqXO
         {
             CPLError(CE_Failure, CPLE_AppDefined,
                      "PDFDoc::displayPageSlice() failed with %s", e.what());
+
+#ifdef USE_OPTCONTENT_HACK
             /* Restore back */
             poCatalog->optContent = poOldOCGs;
+#endif
             delete poSplashOut;
             return CE_Failure;
         }
 
+#ifdef USE_OPTCONTENT_HACK
         /* Restore back */
         poCatalog->optContent = poOldOCGs;
+#endif
 
         SplashBitmap *poBitmap = poSplashOut->getBitmap();
         if (poBitmap->getWidth() != nReqXSize ||
@@ -3734,9 +3748,14 @@ void PDFDataset::ExploreLayersPoppler(GD
                 }
                 else
                     osCurLayer = std::move(osName);
-                // CPLDebug("PDF", "Layer %s", osCurLayer.c_str());
+                    // CPLDebug("PDF", "Layer %s", osCurLayer.c_str());
 
-                OCGs *optContentConfig = m_poDocPoppler->getOptContentConfig();
+#if POPPLER_MAJOR_VERSION > 25 ||                                              \
+    (POPPLER_MAJOR_VERSION == 25 && POPPLER_MINOR_VERSION >= 2)
+                const
+#endif
+                    OCGs *optContentConfig =
+                        m_poDocPoppler->getOptContentConfig();
                 struct Ref r;
                 r.num = poObj->GetRefNum().toInt();
                 r.gen = poObj->GetRefGen();
@@ -3772,11 +3791,19 @@ void PDFDataset::FindLayersPoppler(int i
     if (poPages)
         nPageCount = poPages->GetLength();
 
-    OCGs *optContentConfig = m_poDocPoppler->getOptContentConfig();
+#if POPPLER_MAJOR_VERSION > 25 ||                                              \
+    (POPPLER_MAJOR_VERSION == 25 && POPPLER_MINOR_VERSION >= 2)
+    const
+#endif
+        OCGs *optContentConfig = m_poDocPoppler->getOptContentConfig();
     if (optContentConfig == nullptr || !optContentConfig->isOk())
         return;
 
-    Array *array = optContentConfig->getOrderArray();
+#if POPPLER_MAJOR_VERSION > 25 ||                                              \
+    (POPPLER_MAJOR_VERSION == 25 && POPPLER_MINOR_VERSION >= 2)
+    const
+#endif
+        Array *array = optContentConfig->getOrderArray();
     if (array)
     {
         GDALPDFArray *poArray = GDALPDFCreateArray(array);
@@ -3812,7 +3839,11 @@ void PDFDataset::FindLayersPoppler(int i
 
 void PDFDataset::TurnLayersOnOffPoppler()
 {
-    OCGs *optContentConfig = m_poDocPoppler->getOptContentConfig();
+#if POPPLER_MAJOR_VERSION > 25 ||                                              \
+    (POPPLER_MAJOR_VERSION == 25 && POPPLER_MINOR_VERSION >= 2)
+    const
+#endif
+        OCGs *optContentConfig = m_poDocPoppler->getOptContentConfig();
     if (optContentConfig == nullptr || !optContentConfig->isOk())
         return;
 
diff -rupN --no-dereference gdal-3.10.1-fedora/frmts/pdf/pdfio.cpp gdal-3.10.1-fedora-new/frmts/pdf/pdfio.cpp
--- gdal-3.10.1-fedora/frmts/pdf/pdfio.cpp	2025-01-08 15:54:06.000000000 +0100
+++ gdal-3.10.1-fedora-new/frmts/pdf/pdfio.cpp	2025-02-11 12:05:42.108975633 +0100
@@ -237,23 +237,40 @@ int VSIPDFFileStream::lookChar()
 /*                                reset()                               */
 /************************************************************************/
 
+#if POPPLER_MAJOR_VERSION > 25 ||                                              \
+    (POPPLER_MAJOR_VERSION == 25 && POPPLER_MINOR_VERSION >= 2)
+bool VSIPDFFileStream::reset()
+#else
 void VSIPDFFileStream::reset()
+#endif
 {
     nSavedPos = VSIFTellL(f);
     bHasSavedPos = TRUE;
     VSIFSeekL(f, nCurrentPos = nStart, SEEK_SET);
     nPosInBuffer = -1;
     nBufferLength = -1;
+#if POPPLER_MAJOR_VERSION > 25 ||                                              \
+    (POPPLER_MAJOR_VERSION == 25 && POPPLER_MINOR_VERSION >= 2)
+    return true;
+#endif
 }
 
 /************************************************************************/
 /*                         unfilteredReset()                            */
 /************************************************************************/
 
+#if POPPLER_MAJOR_VERSION > 25 ||                                              \
+    (POPPLER_MAJOR_VERSION == 25 && POPPLER_MINOR_VERSION > 2)
+bool VSIPDFFileStream::unfilteredReset()
+{
+    return reset();
+}
+#else
 void VSIPDFFileStream::unfilteredReset()
 {
     reset();
 }
+#endif
 
 /************************************************************************/
 /*                                close()                               */
diff -rupN --no-dereference gdal-3.10.1-fedora/frmts/pdf/pdfio.h gdal-3.10.1-fedora-new/frmts/pdf/pdfio.h
--- gdal-3.10.1-fedora/frmts/pdf/pdfio.h	2025-01-08 15:54:06.000000000 +0100
+++ gdal-3.10.1-fedora-new/frmts/pdf/pdfio.h	2025-02-11 12:05:42.109184996 +0100
@@ -48,8 +48,20 @@ class VSIPDFFileStream final : public Ba
     virtual int getUnfilteredChar() override;
     virtual int lookChar() override;
 
+#if POPPLER_MAJOR_VERSION > 25 ||                                              \
+    (POPPLER_MAJOR_VERSION == 25 && POPPLER_MINOR_VERSION >= 2)
+    virtual bool reset() override;
+#else
     virtual void reset() override;
+#endif
+
+#if POPPLER_MAJOR_VERSION > 25 ||                                              \
+    (POPPLER_MAJOR_VERSION == 25 && POPPLER_MINOR_VERSION > 2)
+    virtual bool unfilteredReset() override;
+#else
     virtual void unfilteredReset() override;
+#endif
+
     virtual void close() override;
 
     bool FoundLinearizedHint() const
diff -rupN --no-dereference gdal-3.10.1-fedora/frmts/pdf/pdfobject.cpp gdal-3.10.1-fedora-new/frmts/pdf/pdfobject.cpp
--- gdal-3.10.1-fedora/frmts/pdf/pdfobject.cpp	2025-01-08 15:54:06.000000000 +0100
+++ gdal-3.10.1-fedora-new/frmts/pdf/pdfobject.cpp	2025-02-11 12:05:42.109467442 +0100
@@ -979,13 +979,13 @@ class GDALPDFDictionaryPoppler : public
 class GDALPDFArrayPoppler : public GDALPDFArray
 {
   private:
-    Array *m_poArray;
+    const Array *m_poArray;
     std::vector<std::unique_ptr<GDALPDFObject>> m_v{};
 
     CPL_DISALLOW_COPY_ASSIGN(GDALPDFArrayPoppler)
 
   public:
-    GDALPDFArrayPoppler(Array *poArray) : m_poArray(poArray)
+    GDALPDFArrayPoppler(const Array *poArray) : m_poArray(poArray)
     {
     }
 
@@ -1334,7 +1334,7 @@ std::map<CPLString, GDALPDFObject *> &GD
 /*                           GDALPDFCreateArray()                       */
 /************************************************************************/
 
-GDALPDFArray *GDALPDFCreateArray(Array *array)
+GDALPDFArray *GDALPDFCreateArray(const Array *array)
 {
     return new GDALPDFArrayPoppler(array);
 }
diff -rupN --no-dereference gdal-3.10.1-fedora/frmts/pdf/pdfobject.h gdal-3.10.1-fedora-new/frmts/pdf/pdfobject.h
--- gdal-3.10.1-fedora/frmts/pdf/pdfobject.h	2025-01-08 15:54:06.000000000 +0100
+++ gdal-3.10.1-fedora-new/frmts/pdf/pdfobject.h	2025-02-11 12:05:42.109835676 +0100
@@ -405,7 +405,7 @@ class GDALPDFObjectPoppler : public GDAL
     virtual int GetRefGen() override;
 };
 
-GDALPDFArray *GDALPDFCreateArray(Array *array);
+GDALPDFArray *GDALPDFCreateArray(const Array *array);
 
 #endif  // HAVE_POPPLER
 
diff -rupN --no-dereference gdal-3.10.1-fedora/frmts/pdf/pdfsdk_headers_poppler.h gdal-3.10.1-fedora-new/frmts/pdf/pdfsdk_headers_poppler.h
--- gdal-3.10.1-fedora/frmts/pdf/pdfsdk_headers_poppler.h	2025-01-08 15:54:06.000000000 +0100
+++ gdal-3.10.1-fedora-new/frmts/pdf/pdfsdk_headers_poppler.h	2025-02-11 12:05:42.110031975 +0100
@@ -51,10 +51,15 @@
 
 #include <Dict.h>
 
-#define private                                                                \
-    public /* Ugly! Catalog::optContent is private but we need it... */
+#if POPPLER_MAJOR_VERSION > 25 ||                                              \
+    (POPPLER_MAJOR_VERSION == 25 && POPPLER_MINOR_VERSION >= 2)
+#include <Catalog.h>
+#else
+/* Ugly! Catalog::optContent is private but we need it for ancient Poppler versions. */
+#define private public
 #include <Catalog.h>
 #undef private
+#endif
 
 #define private public /* Ugly! PDFDoc::str is private but we need it... */
 #include <PDFDoc.h>
