diff -rupN --no-dereference geos-3.9.1/src/noding/SegmentNodeList.cpp geos-3.9.1-new/src/noding/SegmentNodeList.cpp
--- geos-3.9.1/src/noding/SegmentNodeList.cpp	2021-02-10 19:10:27.000000000 +0100
+++ geos-3.9.1-new/src/noding/SegmentNodeList.cpp	2021-06-17 11:44:40.862229759 +0200
@@ -301,6 +301,8 @@ SegmentNodeList::getSplitCoordinates()
         addEdgeCoordinates(eiPrev, ei, *coordList);
         eiPrev = ei;
     }
+    // Remove duplicate Coordinates from coordList
+    coordList->erase(std::unique(coordList->begin(), coordList->end()), coordList->end());
     return coordList;
 }
 
@@ -312,8 +314,6 @@ SegmentNodeList::addEdgeCoordinates(cons
     createSplitEdgePts(ei0, ei1, pts);
     // Append pts to coordList
     coordList.insert(coordList.end(), pts.begin(), pts.end());
-    // Remove duplicate Coordinates from coordList
-    coordList.erase(std::unique(coordList.begin(), coordList.end()), coordList.end());
 }
 
 
