diff -rupN --no-dereference gdal-3.8.5-fedora/port/cpl_vsil_win32.cpp gdal-3.8.5-fedora-new/port/cpl_vsil_win32.cpp
--- gdal-3.8.5-fedora/port/cpl_vsil_win32.cpp	2024-04-02 12:05:54.000000000 +0200
+++ gdal-3.8.5-fedora-new/port/cpl_vsil_win32.cpp	2024-04-08 09:44:28.226414023 +0200
@@ -812,7 +812,7 @@ int VSIWin32FilesystemHandler::Stat(cons
         }
         else
         {
-            nResult = _wstat64(pwszFilename, pStatBuf);
+            nResult = _wstat64(pwszFilename, reinterpret_cast<struct _stat64 *>(pStatBuf));
             if (nResult < 0)
             {
                 DWORD nLastError = GetLastError();
