diff -rupN geos-3.8.0/src/index/chain/MonotoneChainBuilder.cpp geos-3.8.0-new/src/index/chain/MonotoneChainBuilder.cpp
--- geos-3.8.0/src/index/chain/MonotoneChainBuilder.cpp	2019-10-08 18:20:35.000000000 +0200
+++ geos-3.8.0-new/src/index/chain/MonotoneChainBuilder.cpp	2020-02-20 13:52:22.737581232 +0100
@@ -102,6 +102,7 @@ MonotoneChainBuilder::findChainEnd(const
     const Coordinate* curr = &pts[last];
 
     while(last < npts) {
+        curr = &pts[last];
         // skip zero-length segments, but include them in the chain
         if(!prev->equals2D(*curr)) {
             // compute quadrant for next possible segment in chain
@@ -110,9 +111,8 @@ MonotoneChainBuilder::findChainEnd(const
                 break;
             }
         }
-        ++last;
         prev = curr;
-        curr = &pts[last];
+        ++last;
     }
 #if GEOS_DEBUG
     std::cerr << "MonotoneChainBuilder::findChainEnd() returning" << std::endl;
