diff -rupN geos-3.10.3/include/geos/noding/SegmentNodeList.h geos-3.10.3-new/include/geos/noding/SegmentNodeList.h
--- geos-3.10.3/include/geos/noding/SegmentNodeList.h	2022-06-03 18:04:49.000000000 +0200
+++ geos-3.10.3-new/include/geos/noding/SegmentNodeList.h	2022-06-06 12:01:08.148151403 +0200
@@ -157,9 +157,6 @@ public:
      * Adds an intersection into the list, if it isn't already there.
      * The input segmentIndex is expected to be normalized.
      *
-     * @return the SegmentIntersection found or added. It will be
-     *	   destroyed at SegmentNodeList destruction time.
-     *
      * @param intPt the intersection Coordinate, will be copied
      * @param segmentIndex
      */
diff -rupN geos-3.10.3/include/geos/triangulate/polygon/PolygonEarClipper.h geos-3.10.3-new/include/geos/triangulate/polygon/PolygonEarClipper.h
--- geos-3.10.3/include/geos/triangulate/polygon/PolygonEarClipper.h	2022-06-03 18:04:49.000000000 +0200
+++ geos-3.10.3-new/include/geos/triangulate/polygon/PolygonEarClipper.h	2022-06-06 12:01:52.576152397 +0200
@@ -186,7 +186,6 @@ public:
     *
     * @param polyShell the vertices of the polygon
     * @param triListResult vector to fill in with the resultant Tri s
-    * @return a list of the Tris
     */
     static void triangulate(std::vector<Coordinate>& polyShell, TriList& triListResult);
 
diff -rupN geos-3.10.3/include/geos/triangulate/polygon/VertexSequencePackedRtree.h geos-3.10.3-new/include/geos/triangulate/polygon/VertexSequencePackedRtree.h
--- geos-3.10.3/include/geos/triangulate/polygon/VertexSequencePackedRtree.h	2022-06-03 18:04:49.000000000 +0200
+++ geos-3.10.3-new/include/geos/triangulate/polygon/VertexSequencePackedRtree.h	2022-06-06 12:02:19.176152992 +0200
@@ -137,7 +137,6 @@ public:
     *
     * @param queryEnv the query extent
     * @param result vector to fill with results
-    * @return
     */
     void query(const Envelope& queryEnv, std::vector<std::size_t>& result) const;
 
