diff -rupN --no-dereference geos-3.8.1/CMakeLists.txt geos-3.8.1-new/CMakeLists.txt
--- geos-3.8.1/CMakeLists.txt	2020-03-11 14:56:49.830174310 +0100
+++ geos-3.8.1-new/CMakeLists.txt	2020-03-11 14:56:49.832174317 +0100
@@ -281,7 +281,7 @@ configure_file(cmake/geos-config.cmake
 
 install(TARGETS geos geos_cxx_flags
   EXPORT geos-targets
-  LIBRARY DESTINATION lib${LIB_SUFFIX} NAMELINK_SKIP
+  LIBRARY DESTINATION lib${LIB_SUFFIX}
   ARCHIVE DESTINATION lib${LIB_SUFFIX}
   RUNTIME DESTINATION bin
   INCLUDES DESTINATION include
