diff --git a/loader/Makefile.in b/loader/Makefile.in
index cb8df63..0ab2670 100644
--- a/loader/Makefile.in
+++ b/loader/Makefile.in
@@ -130,7 +130,9 @@ ifdef gtk_build
 	done
 endif
 
-install: installdir install-desktop install-icons
+install-rpm-desktop: install-desktop install-icons
+
+install: installdir
 ifdef gtk_build
 	$(LIBTOOL) --mode=install $(INSTALL) $(SHP2PGSQL-GUI) "$(DESTDIR)$(bindir)/$(SHP2PGSQL-GUI)"
 endif
