diff --git a/GNUmakefile.in b/GNUmakefile.in
index 878548e..3c95a2f 100644
--- a/GNUmakefile.in
+++ b/GNUmakefile.in
@@ -35,7 +35,7 @@ clean-local:
 # TODO: drop 'test' target..
 test: check
 
-check: all docs-check check-no-trailing-blanks
+check: all
 
 check-no-trailing-blanks:
	! find . -name '*.c' -o -name '*.h' -o -name '*.proto' | \
diff --git a/doc/Makefile.in b/doc/Makefile.in
index 31fd778..6268e7f 100644
--- a/doc/Makefile.in
+++ b/doc/Makefile.in
@@ -397,6 +397,9 @@ check: $(XML_INPUTS)
 	$(XMLLINT) --loaddtd --xinclude --valid postgis-out.xml > /dev/null
 endif
 
+check:
+	:
+
 check-localized:
 	@for lang in $(translations); do \
     echo "Creating html for language $$lang..."; \
