diff -Naur proj-4.4.8.orig/nad/test27 proj-4.4.8/nad/test27
--- proj-4.4.8.orig/nad/test27	2004-12-29 13:00:31.428068481 -0800
+++ proj-4.4.8/nad/test27	2004-12-29 13:00:55.742384745 -0800
@@ -8,7 +8,7 @@
 #   Mercator due to greater precision of meridinal distance function.
 #
 OUT=proj_out27
-EXE=../src/proj
+EXE=proj
 INIT_FILE=./nad27
 #
 echo "doing tests into file ${OUT}, please wait"
diff -Naur proj-4.4.8.orig/nad/test83 proj-4.4.8/nad/test83
--- proj-4.4.8.orig/nad/test83	2004-12-29 13:00:31.421069541 -0800
+++ proj-4.4.8/nad/test83	2004-12-29 13:01:04.976985631 -0800
@@ -8,7 +8,7 @@
 #   Mercator due to greater precision of meridinal distance function.
 #
 OUT=proj_out83
-EXE=../src/proj
+EXE=proj
 INIT_FILE=./nad83
 #
 echo "doing tests into file ${OUT}, please wait"
diff -Naur proj-4.4.8.orig/nad/testvarious proj-4.4.8/nad/testvarious
--- proj-4.4.8.orig/nad/testvarious	2004-12-29 13:00:31.420069693 -0800
+++ proj-4.4.8/nad/testvarious	2004-12-29 13:35:39.314658254 -0800
@@ -3,7 +3,7 @@
 #
 #
 OUT=td_out
-EXE=../src/cs2cs
+EXE=cs2cs
 #
 echo "doing tests into file ${OUT}, please wait"
 rm -f ${OUT}
